pkg/controller/job/indexed_job_utils.go

```diff
@@ -201,6 +201,42 @@ func (oi orderedIntervals) has(ix int) bool {
 	return oi[hi].First <= ix
 }
 
+func (oi orderedIntervals) isSubsetOf(superIntervals orderedIntervals) (int, bool) {
+	if len(oi) == 0 && len(superIntervals) == 0 {
+		return 0, true
+	} else if len(oi) == 0 {
+		return superIntervals.total(), true
+	} else if len(superIntervals) == 0 {
+		return oi.total(), true
+	}
+	var diff int
+	for _, sub := range oi {
+		found := false
+		var contains int
+		for _, super := range superIntervals {
```

Comment:
Isn't this actually `O(n^2)` as for every `sub` we run through entire `superIntervals`?
I think we could keep an index for iterating over `superIntervals`, because we never need to look back. This way we would only do one pass for `oi` and one pass for `superIntervals`

Reply:
> I think we could keep an index for iterating over superIntervals, because we never need to look back. This way we would only do one pass for oi and one pass for superIntervals

You're right. I'm not sure by myself why I implemented such a poor approach :(
The current implementation is O(n*m).
If we select the following implementation, the below has O(n+m) where the interval of `First` and `Last` is small enough.

Given that the `.spec.completions` is small, O(n*m) is better than O(n+m), but I understand that we should consider the case of large `.spec.completions`. So, let me replace it with the O(n+m) approach.

@mimowo Does it make sense?

```go
...
	superElements := sets.New[int]()
	for i := range superIntervals {
		for j:=superIntervals[i].First; j <= superIntervals[i].Last; j++ {
			superElements.Insert(j)
		}
	}
	var diff int
	for _, sub := range oi {
		for i:=sub.First; i<=sub.Last; i++ {
			if !superElements.Has(i) {
				diff++
			}
		}
	}
...
```

Reply:
I talked with @mimowo offline, and then we decided to select another better approach.

Reply:
@mimowo and me synced our consideration offline.

In conclusion, we decided to implement a dedicated function only for successPolicy, not considering generic use cases for the orderedIntervals because making it generic would bring it more complexity, and this `isSubsetOf` wouldn't be used in other cases excluding successPolicy.

Reply:
Done.